web/client/api/Model.js

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2024, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import axios from 'axios';
+import proj4 from 'proj4';
+
+/**
+ * get ifc model main info such as: longitude, latitude, height and scale
+ * and additional info in case of existing a specified crs like: projectedCrs
+ * @param {object} ifcData includes 2 items: ifcApi, WebIFC
+ * @param {string} modelVersion the version of ifc file [e.g: IFC2x3, IFC4, IFC4x3]
+ * @param {number} modelID it is the model if of the ifc model e.g: 0,1,2
+ * @return {object} return originProperties that includes: latitude, longitude, height, scale and "projectedCrs" in case of georeferenced IFC4 model
+ * @
+ */
+// extract model origin point
+function getModelOriginCoords(ifcData, modelVersion, modelID) {
+    const { ifcApi, WebIFC } = ifcData;
+    let originProperties = {};
+    if (modelVersion?.includes("IFC4")) {
+        let projectedCrs;
+        let mapConversion;
+        let projectedCrsLineId = ifcApi.GetLineIDsWithType(modelID, WebIFC.IFCPROJECTEDCRS);    // eslint-disable-line
+        let mapConversionLineId = ifcApi.GetLineIDsWithType(modelID, WebIFC.IFCMAPCONVERSION);    // eslint-disable-line
+        if (projectedCrsLineId.size()) {
+            let typeID = projectedCrsLineId.get(0);
+            let projectedCrsObj = ifcApi.GetLine(modelID, typeID);    // eslint-disable-line
+            projectedCrs = projectedCrsObj?.Name?.value;
+        }
+        if (mapConversionLineId.size()) {
+            let typeID = mapConversionLineId.get(0);
+            let mapConversionObj = ifcApi.GetLine(modelID, typeID);    // eslint-disable-line
+            mapConversion = {
+                northings: mapConversionObj?.Northings?.value,          // x coord
+                eastings: mapConversionObj?.Eastings?.value,            // y coord
+                orthogonalHeight: mapConversionObj?.OrthogonalHeight?.value,    // height (z coord)
+                xAxisOrdinate: mapConversionObj?.XAxisOrdinate?.value,
+                xAxisAbscissa: mapConversionObj?.XAxisAbscissa?.value,
+                rotation: Math.atan2(mapConversionObj?.XAxisOrdinate?.value || 0, mapConversionObj?.XAxisAbscissa?.value || 0) * 180.0 / Math. PI,
```

Comment:
With the tested ifc model with projected it seems that there is something wrong with the rotation. The formula seems correct but the model is not aligned with the building represented on map.
For the moment my proposal is to add in the ui the management of heading property so we could adjust this. Ideally we need more model to test also from other sources